src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnDecoder.GeneralizedTime.cs

```diff
@@ -125,12 +117,12 @@ public DateTimeOffset ReadGeneralizedTime(Asn1Tag expectedTag, bool disallowFrac
             if (strict && contentOctets.Length < 15)
             {
                 // yyyyMMddHHmmssZ
-                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                throw new AsnContentException(SR.ContentException_InvalidUnderCerOrDer_TryBer);
             }
             else if (contentOctets.Length < 10)
             {
                 // yyyyMMddHH
-                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                throw new AsnContentException();
```

Comment:
Just curious, what warrants a custom message above but not for this one?

Reply:
This throw is "the data is so small there's no chance it's legal", so it just gets the general "I don't understand this content, it's not a legal payload" message.  The previous case is "I'm only throwing here because of a mode the reader is in", so it throws with a "this might work in a different mode"-type message.